File: 6-ZigzagConversion.js
console.log('starting')

var convert = function(s, numRows) {
    if (numRows==0){
        return s;
    }
    let result = '';
    let word=[];
    let direction = 1;
    for (let j=0; j<numRows; j++){
        word[j]='';
    }
    let i=0;
    let j=0;
    while (i<s.length){
        word[j]+=s[i]
        j+=direction;
        if(j>=numRows){
            direction = -1;
            j =numRows-2;
        }
        if(j<=0){
            direction = 1;
            j = 0;
        }   
        i++
    }

    for (let j=0; j<word.length; j++){
        if(typeof word[j] != 'undefined'){

            result+=word[j];
        }
    }
    return result;
};

console.log('result',convert('PAYPALISHIRING',3));
console.log('PINALSIGYAHRPI');

File: python/6_Zigzag_Conversion.py
class Solution(object):
    def convert(self, s, numRows):
        """
        :type s: str
        :type numRows: int
        :rtype: str
        """
        if numRows ==1: return s
        row = 1
        goingUp = False
        working = True
        rows = []
        result = ''
        for i in range(numRows):
            rows.append('')
        for letter in s:
            rows[row-1]+=letter
            if row == numRows or goingUp:
                goingUp = True
                row -=1
                if row ==1:
                    goingUp = False
            else:
                row +=1
        for group in rows:
            result += group
        return result

tests = []
results = []

tests.append(["ABC",1])
results.append("ABC")

tests.append(["PAYPALISHIRING",3])
results.append("PAHNAPLSIIGYIR")

tests.append(["PAYPALISHIRING",4])
results.append("PINALSIGYAHRPI")

for test in range(len(tests)):
    result = Solution().convert(tests[test][0],tests[test][1])
    print('This is test ' + str(test),'Result',result,results[test])